src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs

```diff
@@ -905,7 +934,8 @@ internal static bool RequiresExplicitOverride(this MethodSymbol method)
                 // We can ignore interface implementation changes since the method is already metadata virtual (since override).
                 // TODO: do we want to add more sophisticated handling for the case where there are multiple runtime-overridden methods?
                 MethodSymbol runtimeOverriddenMethod = method.GetFirstRuntimeOverriddenMethodIgnoringNewSlot(ignoreInterfaceImplementationChanges: true);
-                return csharpOverriddenMethod != runtimeOverriddenMethod &&
+                return runtimeOverriddenMethod is null ||
+                    csharpOverriddenMethod != runtimeOverriddenMethod &&
                     method.IsAccessor() != runtimeOverriddenMethod.IsAccessor();
```

Comment:
>method.IsAccessor() != runtimeOverriddenMethod.IsAccessor() [](start = 20, length = 59)

It is not obvious that this condition should still be required for a covariant return case.

Reply:
I suspect you're misreading the precedence.  The first condition `runtimeOverriddenMethod is null` is for the covariant return case. The remainder is the old condition.  There are tests for both.  Either is enough to require a methodimpl.  This expression is rewritten for clarity in [https://github.com/dotnet/roslyn/pull/44025](https://github.com/dotnet/roslyn/pull/44025)

---
In reply to: [421739206](https://github.com/dotnet/roslyn/pull/43797#discussion_r421739206) [](ancestors = 421739206)